clang-ocl.in

```diff
@@ -32,8 +33,22 @@ do
     esac
     shift
 done
- 
- 
+
+if [-e $BITCODE_DIR/opencl.amdgcn.bc ]; then
+  BITCODE_PREF="-Xclang -mlink-bitcode-file -Xclang"
+  BITCODE_ARGS="-nogpulib \
+    $BITCODE_PREF $BITCODE_DIR/opencl.amdgcn.bc \
+    $BITCODE_PREF $BITCODE_DIR/ocml.amdgcn.bc \
+    $BITCODE_PREF $BITCODE_DIR/ockl.amdgcn.bc \
+    $BITCODE_PREF $BITCODE_DIR/oclc_correctly_rounded_sqrt_off.amdgcn.bc \
+    $BITCODE_PREF $BITCODE_DIR/oclc_daz_opt_on.amdgcn.bc \
+    $BITCODE_PREF $BITCODE_DIR/oclc_finite_only_off.amdgcn.bc \
+    $BITCODE_PREF $BITCODE_DIR/oclc_isa_version_${gfxip}.amdgcn.bc \
+    $BITCODE_PREF $BITCODE_DIR/oclc_unsafe_math_off.amdgcn.bc"
+else
+  BITCODE_ARGS="--rocm-path=${ROCM_PATH}"
```

Comment:
`ROCM_PATH` is not configured in the cmake at all.

It also seems to be a very ambiguous flag. The location of device libs is not the same location as rocm.

Reply:
Hi Paul, it seems that the new layout plans to install the device libs to /opt/rocm/amdgcn/bitcode/. I think for the newer compiler onwards, the install location is /opt/rocm/amdgcn, using whichever ROCM_PATH is given. I will check the cmake to make sure its configured properly

Reply:
Dont we need to do `find_package(AMDDeviceLibs PATHS /opt/rocm)`? If device libs is not installed in /opt/rocm, then the user would set `CMAKE_PREFIX_PATH` to its installed location, and `find_package` will find it. I believe it sets the `AMD_DEVICE_LIBS_PREFIX` variable to the path we need for `--rocm-path`.  Hopefully, this is only used for device libs and not other rocm components(which the user could install in other locations).

Reply:
Do you know if BITCODE_DIR is valid? Because we can skipp the rocm-path, and use bitcode-dir to set a different flag in the compiler. Something like: --hip-device-lib-path=$BITCODE_DIR works in our AMD toolchain

Reply:
Ideally, `find_package(AMDDeviceLibs)` should provide this `BITCODE_DIR` path because it can change depending on the version. Also, does `hip-device-lib-path` work for opencl?

Reply:
Sorry, I need to update the logic in CMakeLists.txt for BITCODE_DIR. We can use that instead.

Reply:
@arsenm - do you know if the --hip-device-lib-path works for OpenCL same as the way it works for HIP-Clang? (such as linking with the bitcodes at amdgcn/bitcode)

Reply:
It seems like the CMakeLists.txt only works for hcc's clang. Our llvm amdgpu compiler needs to provide a `LLVMAMDGPUConfig.cmake` file so we can find the clang that supports amdgpu.

Reply:
Or we can just use the `CMAKE_CXX_COMPILER` for now.

Reply:
I think the CI system might install AMDDeviceLibs to a different location than the default. I've removed the ROCM_PATH and used the BITCODE_DIR that is passed in.